### ISSUE TITLE
Playback interrupted during track

### ISSUE THREAD

**Issue by burnoutberni** [librespot-org/librespot#168]
Hi, I just encountered the following bug:

I was playing some music using librespot on a RPi3, when suddenly it was interrupted without any further notice and I was unable to resume playback, also after trying to reconnect to the librespot client from another Spotify client. Only after restarting the librespot service, I was able to use it again.

Please find below the relevant systemd log:

```
Feb 24 14:08:26 raspberrypi librespot[1380]: INFO:librespot: librespot 685fb4e (2018-02-22). Built on 2018-02-23. Build ID: TxmbQHxj
[…]
Feb 26 10:53:51 raspberrypi librespot[1380]: INFO:librespot_playback::player: Track "Digital Lion" loaded
Feb 26 10:54:04 raspberrypi librespot[1380]: ERROR:librespot_core::channel: channel error: 2 1
Feb 26 10:54:04 raspberrypi librespot[1380]: WARN:librespot_audio::fetch: error from channel
```

**Comment by burnoutberni** [comment]
Not sure if it's the same bug, but I just experienced a similar issue on another system (x86_64, Arch Linux, librespot built for pulseaudio). Once again, interrupting halfway during playback of a track, needs restart to work again.

```
Feb 27 13:12:57 tini-x230 librespot[2026]: INFO:librespot: librespot 593dfa0 (2018-02-27). Built on 2018-02-27. Build ID: h7AH3J2x
[…]
Feb 27 16:17:11 tini-x230 librespot[2026]: INFO:librespot_playback::player: Loading track "El Pueblo Unido"
Feb 27 16:17:15 tini-x230 librespot[2026]: INFO:librespot_playback::player: Track "El Pueblo Unido" loaded
Feb 27 16:17:54 tini-x230 librespot[2026]: thread 'main' panicked at 'Box<Any>', core/src/session.rs:79:43
Feb 27 16:17:54 tini-x230 librespot[2026]: stack backtrace:
Feb 27 16:17:54 tini-x230 librespot[2026]:    0:     0x55bfa742b52b - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::hb98fbe643b37b8bb
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
Feb 27 16:17:54 tini-x230 librespot[2026]:    1:     0x55bfa74359ce - std::sys_common::backtrace::print::ha7844e468550fcb9
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/sys_common/backtrace.rs:68
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/sys_common/backtrace.rs:57
Feb 27 16:17:54 tini-x230 librespot[2026]:    2:     0x55bfa74244d0 - std::panicking::default_hook::{{closure}}::h83c090f00cd2917d
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panicking.rs:381
Feb 27 16:17:54 tini-x230 librespot[2026]:    3:     0x55bfa742400c - std::panicking::default_hook::hf9722061a353cd29
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panicking.rs:397
Feb 27 16:17:54 tini-x230 librespot[2026]:    4:     0x55bfa742491b - std::panicking::rust_panic_with_hook::h574be4fada9826dc
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panicking.rs:577
Feb 27 16:17:54 tini-x230 librespot[2026]:    5:     0x55bfa7169ee2 - std::panicking::begin_panic::h208cb492118f3ce1
Feb 27 16:17:54 tini-x230 librespot[2026]:    6:     0x55bfa71c9d23 - <futures::future::map_err::MapErr<A, F> as futures::future::Future>::poll::h217>
Feb 27 16:17:54 tini-x230 librespot[2026]:    7:     0x55bfa740962f - futures::task_impl::std::set::h96e54a5baac7defc
Feb 27 16:17:54 tini-x230 librespot[2026]:    8:     0x55bfa7408175 - <futures::task_impl::Spawn<T>>::poll_future_notify::hef372e91815853df
Feb 27 16:17:54 tini-x230 librespot[2026]:    9:     0x55bfa740cbf1 - tokio_core::reactor::Core::poll::h12105a721cbed6b5
Feb 27 16:17:54 tini-x230 librespot[2026]:   10:     0x55bfa7055b9c - tokio_core::reactor::Core::run::h83ad52697f079d3a
Feb 27 16:17:54 tini-x230 librespot[2026]:   11:     0x55bfa704832f - librespot::main::hd7f85a5a09f0fc9a
Feb 27 16:17:54 tini-x230 librespot[2026]:   12:     0x55bfa70566a2 - std::rt::lang_start::{{closure}}::h615c01857cac4f3a
Feb 27 16:17:54 tini-x230 librespot[2026]:   13:     0x55bfa74245d7 - std::panicking::try::do_call::h0b062116b4efed59
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/rt.rs:59
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panicking.rs:480
Feb 27 16:17:54 tini-x230 librespot[2026]:   14:     0x55bfa7449ece - __rust_maybe_catch_panic
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libpanic_unwind/lib.rs:101
Feb 27 16:17:54 tini-x230 librespot[2026]:   15:     0x55bfa742b249 - std::rt::lang_start_internal::h6bcde385b3ee3341
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panicking.rs:459
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/panic.rs:365
Feb 27 16:17:54 tini-x230 librespot[2026]:                                at /checkout/src/libstd/rt.rs:58
Feb 27 16:17:54 tini-x230 librespot[2026]:   16:     0x55bfa7049143 - main
Feb 27 16:17:54 tini-x230 librespot[2026]:   17:     0x7fd1011e6f49 - __libc_start_main
Feb 27 16:17:54 tini-x230 librespot[2026]:   18:     0x55bfa703a2f9 - _start
Feb 27 16:17:54 tini-x230 librespot[2026]:   19:                0x0 - <unknown>
```

**Comment by sashahilton00** [comment]
The bug is the same as the one in #103, currently being tracked in #134. It requires a substantial rewrite of how sessions are handled, and hence hasn't been implemented yet. IT happens when the TCP connection to Spotify is lost, most likely your internet drooped briefly at some point

**Comment by burnoutberni** [comment]
Ah, thanks for the explanation and sorry for double posting the same bug.

**Comment by vikeri** [comment]
Thanks for a very cool library! Keep up the good work!

However, I'm also have this issue all the time, possibly since my Android phone disconnects from WiFi when it goes idle after a while. Any way of hacking a solution while waiting for #134?
```
Mar 09 22:49:20 raspberrypi librespot[6525]: thread 'main' panicked at 'Box<Any>', core/src/session.rs:79:43
Mar 09 22:49:20 raspberrypi librespot[6525]: stack backtrace:
Mar 09 22:49:20 raspberrypi librespot[6525]: 0:   0x92363f - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 1:   0x92d89b - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 2:   0x91bec7 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 3:   0x91b9a3 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 4:   0x91c373 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 5:   0x62ddeb - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 6:   0x6a6d7f - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 7:   0x8fd61f - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 8:   0x8fbecf - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 9:   0x900bab - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 10:   0x517ec3 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 11:   0x508aa3 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 12:   0x518a13 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 13:   0x91c01b - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 14:   0x942c33 - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 15:   0x9232cb - <unknown>
Mar 09 22:49:20 raspberrypi librespot[6525]: 16:   0x509583 - main
Mar 09 22:49:20 raspberrypi librespot[6525]: 17: 0x76cca293 - __libc_start_main
```

**Comment by ComlOnline** [comment]
This shouldn't be related to your phone as libresport connects direct to Spotify and the bug referenced is when a connect is lost between librespot and Spotify. Saying that your also have a stripped binary, hence when you have a lot of `<unkown>`s. Could you open a new issue?

**Comment by vikeri** [comment]
Refiled issue as #186
